### PR TITLE
Fix unused parameters in `proxy/http3` functionality

### PR DESCRIPTION
**Pull request by freak82**
Fix unused parameters using `/* name ATS_UNUSED */`

This pull request is part of the effort for [removing](https://github.com/apache/trafficserver/issues/11377) the `-Wno-unused-parameter` warning suppression.

### CODE REVIEW

**Comment by JosiahWI** [comment]
[approve ci rocky]

File: src/proxy/http3/test/test_QPACK.cc [hunk 1]
```diff
@@ -329,7 +329,8 @@ test_encode(const char *qif_file, const char *out_file, int dts, int mbs, int am
 }
 
 static int
-test_decode(const char *enc_file, const char *out_file, int dts, int mbs, int am, const char *app_name)
+test_decode(const char *enc_file, const char *out_file, int dts, int mbs, int /* am ATS_UNUSED */,
```
Comment (JosiahWI) [inline src/proxy/http3/test/test_QPACK.cc:332]: Please remove the `am` and `app_name` parameters completely.

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Looks good. Let's remove the unused parameters in the QPACK test. The ones in the QPACK implementation are needed later according to comments.

**Comment by freak82** [comment]
You are right. I should have left these changes for tomorrow. To do them on a clear head.
I'm about to fix it. 
And the other pull request is probably broken too.

**Comment by JosiahWI** [comment]
The AuTest chunked_encoding_disabled failed.

```
           -       [DEBUG]: Response connection: keep-alive
           +       [DEBUG]: Response contained "Connection: close". Closing the connection for key: 1.
           +       [DEBUG]: Reading response header.
           +       [DEBUG]: Received an HTTP/1 200 response for key 2 with headers:
```

**Comment by JosiahWI** [comment]
[approve ci autest]

**Review by JosiahWI — APPROVED** [review]
Thanks for the cleanup!

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**